### PR TITLE
Checks file exists on `Dir::make()`

### LINKED ISSUE #3442: Dir::make() emits E_WARNING, if a file with the same name exists
**Describe the bug**  
The `Kirby\Toolkit\Dir::make()` has some sanity checks for preventing errors, e.g. checking the directory already exists and throws an exception, if it cannot be created. But if a file with the same name exists, where the directory should be created, these checks don’t work with the current implementation.

**To Reproduce**  
1. Enable debugging mode
2. Create an example directory with a file in it: `media/example/testfile`
3. Call `Dir::make('media/example/testfile')`
4. The method will emit an `E_WARNING` level error (see <https://www.php.net/manual/en/function.mkdir.php#refsect1-function.mkdir-errors>)

**Expected behavior**  
Other checks are throwing an `Exception`, so I guess as directory creation also fails in this case, it should also throw an Exception instead of emitting an error.

**Kirby Version**  
3.5.6

**Console output**  
```
Warning: mkdir(): File exists
```

**Additional context**  
PHP 8.0.7 on some Linux server

### PR DESCRIPTION
**Pull request by afbora**
## Describe the PR

Implemented @distantnative suggestion: https://github.com/getkirby/kirby/issues/3442#issuecomment-864388449

### Fixes
- Fixed warning when there is a file in the creating folder name #3442 

## Breaking changes

None

## Related issues/ideas

- Fixes #3442 

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [x] Add changes to release notes draft in Notion

### CODE REVIEW

File: src/Toolkit/Dir.php [hunk 1]
```diff
@@ -221,6 +221,7 @@ public static function link(string $source, string $link): bool
      * @param string $dir The path for the new directory
      * @param bool $recursive Create all parent directories, which don't exist
      * @return bool True: the dir has been created, false: creating failed
+     * @throws \Exception
```
Comment (lukasbestle) [inline src/Toolkit/Dir.php]:
```suggestion
     * @throws \Exception If a file with the provided path already exists or the parent directory is not writable
```

File: tests/Toolkit/DirTest.php [hunk 2]
```diff
@@ -158,6 +158,17 @@ public function testMake()
         $this->assertFalse(Dir::make(''));
     }
 
+    public function testMakeFileExists()
+    {
+        $test = $this->tmp . '/test';
+
+        $this->expectExceptionMessage('Expection');
```
Comment (lukasbestle) [inline tests/Toolkit/DirTest.php]:
```suggestion
        $this->expectException('Expection');
```

**Comment by bastianallgeier** [comment]
@afbora I'm happy with the implementation. Could you fix the remaining issues and @lukasbestle's suggestion?

**Comment by afbora** [comment]
@bastianallgeier AFK for now, I'll deal afternoon

**Review by bastianallgeier — APPROVED** [review]

**Comment by bastianallgeier** [comment]
@afbora I quickly fixed it for you with Github's suggested commits feature. I should use it more often :) It's pretty great.

**Pull request merged**